tests/test_oneclick.py

```diff
@@ -0,0 +1,295 @@
+            oneclick_api=self.api
+        )
+        assert isinstance(script.provider, 
+                          OneClickBibliographicCoverageProvider)
+        eq_(script.provider.api, self.api)
+
+
+    def test_invalid_or_unrecognized_guid(self):
+        # A bad or malformed GUID can't get coverage.
+
+        identifier = self._identifier()
+        identifier.identifier = 'bad guid'
+        
+        error = '{"errorCode": "InvalidGuid", "message": "An invalid guid was given.", "token": "7aebce0e-2e88-41b3-b6d3-82bf15f8e1a2"}'
+        self.api.queue_response(200, content=error)
+
+        failure = self.provider.process_item(identifier)
+        assert isinstance(failure, CoverageFailure)
+        eq_(False, failure.transient)
+        eq_("ISBN [bad guid] is not in library [library_id_123]'s catalog.", failure.exception)
+
+        # This is for when the GUID is well-formed but doesn't
+        # correspond to any real Overdrive book.
+        error = '{"errorCode": "NotFound", "message": "Not found in Overdrive collection.", "token": "7aebce0e-2e88-41b3-b6d3-82bf15f8e1a2"}'
+        self.api.queue_response(200, content=error)
+
+        failure = self.provider.process_item(identifier)
+        assert isinstance(failure, CoverageFailure)
+        eq_(False, failure.transient)
+        eq_("ISBN [bad guid] is not in library [library_id_123]'s catalog.", failure.exception)
+
+
+    def test_process_item_creates_presentation_ready_work(self):
+        # Test the normal workflow where we ask Axis for data,
+        # Axis provides it, and we create a presentation-ready work.
+        
+        datastr, datadict = self.get_data("response_isbn_found_1.json")
+        self.api.queue_response(200, content=datastr)
+        
+        # Here's the book mentioned in response_isbn_found_1.
+        identifier = self._identifier(identifier_type=Identifier.ONECLICK_ID)
+        identifier.identifier = '9780307378101'
+
+        # This book has no LicensePool.
+        eq_(None, identifier.licensed_through)
+
+        # Run it through the Axis360BibliographicCoverageProvider
+        provider = OneClickBibliographicCoverageProvider(
+            self._db, oneclick_api=self.api
+        )
+        result = provider.process_item(identifier)
+        eq_(identifier, result)
+
+        # A LicensePool was created. But we do NOT know how many copies of this
+        # book are available, and what formats it's available in.
+        pool = identifier.licensed_through
+        eq_(0, pool.licenses_owned)
+        [lpdm] = pool.delivery_mechanisms
+        eq_('application/epub+zip (OneClick DRM)', lpdm.delivery_mechanism.name)
+
+        # A Work was created and made presentation ready.
+        eq_('Tea Time for the Traditionally Built', pool.work.title)
+
+        # TODO: determine what endpoints we want to call to get the licensing info 
+        # and make the work presentation-ready.
+        #eq_(True, pool.work.presentation_ready)
+       
```

Comment:
The work should be made presentation ready once process_item() finishes. A work is made presentation ready once we have enough bibliographic information. There has to be _a_ license pool, but that pool doesn't have to actually have any copies. See test_overdrive.py for an example.